tools/build.sh

```diff
@@ -1,3 +1,12 @@
-dirname=$(dirname $0)
-cd "$dirname/.."
-./node_modules/.bin/doxx "$(pwd)/config/doxx"
+#!/usr/bin/env bash
+
+# exit on error, unassigned vars, or pipe fails
+set -euo pipefail
+
+# get the absolute path to the script in case it is called from elsewhere
+dirname="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
> dirname

_Generally_ not a good idea to have a variable named the same as a command. I know you've carried it over from earlier code. Please rename if you get a chance.

Reply:
+1 to this and I should have caught it, a good term for this path is `SCRIPT_DIR` or something like that

Reply:
Roger that.

Reply:
> @vipulgupta2048 Now that I have your attention stuck_out_tongue, please do share what the whitespaces in the command `$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"` are for. Althought they shouldn't make any difference at runtime, it is not customary to have such spaces before/after commands. They made me wonder whether they are there to address some weird shell corner case (are they!?).

That was me, I recommended that change to the script to safely determine the script path and I copy-pasted the solution from SO since I had used it previously and I vaguely remembered what it looked like.

The spaces can be removed and it only adds to the readability in this case.

Reply:
Got it 👍🏻 
Thanks @klutchell .

Reply:
All suggestions added, thanks for the review. Resolving.